### PR TITLE
Application overview modal

### PR DESCRIPTION
**Pull request by abhinandan13jan**
## Fixes 

https://issues.redhat.com/browse/HACBS-1148

## Description

- adds `application overview modal` 
- Uses localStorage to set user close activity and only launches the modal on first visit.
- adds`learn more` link to launch modal

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

Screenshot 2022-10-13 at 8 00 32 PM
Screenshot 2022-10-13 at 8 00 45 PM

## How to test or reproduce?

Yet to add unit tests

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/hacbs/components/ApplicationDetails/HacbsApplicationDetails.tsx [hunk 1]
```diff
@@ -22,6 +23,10 @@ type HacbsApplicationDetailsProps = {
 };
 
 const HacbsApplicationDetails: React.FC<HacbsApplicationDetailsProps> = ({ applicationName }) => {
+  const [showApplicationModal, setShowApplicationModal] = React.useState<boolean>(
+    !window.localStorage.getItem(HACBS_APPLICATION_MODAL_KEY),
```
Comment (christianvogt) [inline src/hacbs/components/ApplicationDetails/HacbsApplicationDetails.tsx:27]:
```suggestion
    window.localStorage.getItem(HACBS_APPLICATION_MODAL_KEY) === 'true',
```

File: src/hacbs/components/ApplicationDetails/ApplicationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  showApplicationModal: boolean;
+  onClose: () => void;
+}
+
+const ApplicationModal: React.FC<ApplicationModalProps> = ({ showApplicationModal, onClose }) => {
+  const footer = (
+    <div className="pf-u-display-flex pf-u-justify-content-center pf-u-flex-fill">
+      <Button
+        data-testid="hacbs-getting-started-modal-dismiss"
+        aria-label="get started"
```
Comment (christianvogt) [inline src/hacbs/components/ApplicationDetails/ApplicationModal.tsx:17]: Wrong label?
I don't think an aria label is necessary here as the button itself has valid text.

File: src/hacbs/components/ApplicationDetails/ApplicationModal.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        onClick={onClose}
+      >
+        Got it, thanks
+      </Button>
+    </div>
+  );
+
+  return (
+    <Modal
+      aria-label="commits-getting-started-modal"
```
Comment (christianvogt) [inline src/hacbs/components/ApplicationDetails/ApplicationModal.tsx:27]: Aria labels should be human readable text as screen readers will read this value.

File: src/hacbs/components/ApplicationDetails/ApplicationModal.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      width="fit-content"
+      appendTo={document.querySelector('#hacDev-modal-container') as HTMLElement}
```
Comment (christianvogt) [inline src/hacbs/components/ApplicationDetails/ApplicationModal.tsx:29]: Oh boy, i never realized this is being done in this manner everywhere. 

This should be made into a reusable utility so it's shared code.

File: src/hacbs/components/ApplicationDetails/ApplicationModal.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      footer={footer}
+      isOpen={showApplicationModal}
+      showClose={false}
+      data-testid="hacbs-application-modal"
+    >
+      <div className="pf-u-display-flex pf-u-justify-content-center pf-u-flex-fill">
+        <img src={imageUrl} alt="getting started with commits" />
```
Comment (christianvogt) [inline src/hacbs/components/ApplicationDetails/ApplicationModal.tsx:36]: wrong alt text

File: src/hacbs/components/ApplicationDetails/ApplicationModal.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      </div>
+      <div className="pf-u-display-flex pf-u-justify-content-center pf-u-flex-fill">
+        <TextContent>
+          <Text className="pf-u-text-align-center" component={TextVariants.h2}>
+            Configure your application pipeline
+          </Text>
+          <div>Configure your app a pipeline by adding these pieces.</div>
```
Comment (christianvogt) [inline src/hacbs/components/ApplicationDetails/ApplicationModal.tsx:43]: Grammar issue. Even though this was copied directly from the UXD doc :)

**Comment by abhinandan13jan** [comment]
@christianvogt updated

**Comment by karthikjeeyar** [comment]
image

Adding lgtm but @Ranelim  we may need a different image without the blue square below the selected node, don't know if this is intentional.

/lgtm

**Pull request merged**